modules/lib/src/main/resources/assets/js/app/ui2/dialog/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,133 @@
+import {Button, Dialog, cn} from '@enonic/ui';
+import type {ComponentPropsWithoutRef, ReactElement, ReactNode} from 'react';
+import {createContext, useCallback, useContext, useMemo, useState} from 'react';
+import {i18n} from '@enonic/lib-admin-ui/util/Messages';
+import {forwardRef} from 'react';
+
+//
+// Confirmation dialog context
+//
+
+type ConfirmationDialogContextValue = {
+    unlocked: boolean;
+    setUnlocked: (unlocked: boolean) => void;
+};
+
+const ConfirmationDialogContext = createContext<ConfirmationDialogContextValue | null>(null);
+
+const useConfirmationDialog = (): ConfirmationDialogContextValue => {
+    const ctx = useContext(ConfirmationDialogContext);
+    if (!ctx) {
+        // Safe default: unlocked to keep buttons enabled if used outside Content wrapper
+        return {unlocked: true, setUnlocked: () => void 0};
+    }
+    return ctx;
+};
+
+type ConfirmationDialogProviderProps = {
+    defaultUnlocked?: boolean;
+    children?: ReactNode;
+};
+
+const ConfirmationDialogProvider = ({defaultUnlocked = true, children}: ConfirmationDialogProviderProps): ReactElement => {
+    const [unlocked, setUnlocked] = useState<boolean>(defaultUnlocked);
+
+    const value = useMemo<ConfirmationDialogContextValue>(() => ({unlocked, setUnlocked}), [unlocked]);
+
+    return <ConfirmationDialogContext.Provider value={value}>{children}</ConfirmationDialogContext.Provider>;
+};
+
+//
+// Content: wraps Dialog.Content with provider to keep shared state
+//
+
+export type ConfirmationDialogContentProps = {
+    defaultUnlocked?: boolean;
+    className?: string;
+    children?: ReactNode;
+} & ComponentPropsWithoutRef<typeof Dialog.Content>;
+
+const ConfirmationDialogContent = forwardRef<HTMLDivElement, ConfirmationDialogContentProps>(
+    ({defaultUnlocked = true, className, children, ...props}, ref): ReactElement => {
+        return (
+            <Dialog.Content ref={ref}
+                            className={cn('!outline-none [&_button]:!outline-none gap-3 min-w-170 max-w-fit', className)} {...props}>
+                <ConfirmationDialogProvider defaultUnlocked={defaultUnlocked}>{children}</ConfirmationDialogProvider>
+            </Dialog.Content>
+        );
+    }
+);
+ConfirmationDialogContent.displayName = 'ConfirmationDialog.Content';
+
+//
+// Footer: renders Cancel/Confirm actions and respects unlocked state
+//
+
+type ButtonClick = NonNullable<ComponentPropsWithoutRef<'button'>['onClick']>;
+
+export type ConfirmationDialogFooterProps = {
+    className?: string;
+    cancelLabel?: ReactNode;
+    submitLabel?: ReactNode;
+    onCancel?: ButtonClick;
+    onConfirm?: ButtonClick;
+} & ComponentPropsWithoutRef<'footer'>;
+
+const ConfirmationDialogFooter = ({
+                                      className,
+                                      cancelLabel = i18n('action.cancel'),
+                                      submitLabel = i18n('action.submit'),
```

Comment:
1. Move initialization to function's body.
2. Use `useI18n` hook instead of that function. Even though the hook is just alias, it will be replaced with an actual hook in the future.